### DISCUSSION TITLE
"Proces-Simulator" run a process with given instance variables

### DISCUSSION THREAD

**Discussion by hairpin666** [flowsforapex/apex-flowsforapex#573] [Q&A] [answered]
Hi,

First of all, thank you for open sourcing this project.
We might have a use-case where we want to "simulate" on of our processes. The process itselve calculates a rating based on some given data (e.g. some risk factors). To make the process more transparent we want to draw it up as a model and show the user which branch the process went with the given risk factors. The process modelling and the instance variables are setup. 
I tried to create a page which contains all instance variable and set them with the Instance Variable Plugin.
We can call the manage flow instance step plugin to go to the next step, all good :). 

I'd like to tell the engine to go throug ALL steps at once as far as possible not only one step at a time. Is there a way to acomplish that with the plugin or a PL/SQL call?

Thanks and best regards!

**Comment by rallen2010** [comment, marked as answer]
The engine is only going to stop for tasks which require human input - so Task, userTask and manualTask.   Where you start building your model using standard Tasks, as you evolve your design into an executable model, most of your plain tasks should get changed into userTask, serviceTask, scriptTask, businessRule Task, etc -- and all of those except userTask and ManualTask will flow from one task to the next.

Not sure if that answers the question, or if I fully understand the question.   Hope to do a quick call if you need help.

**Reply by hairpin666** [reply]
Thanks, works perfectly with a serviceTask.

**Comment by hairpin666** [comment]
Hi Richard,

Thanks for your answer and your offer for a call. I will have a look into changing the taks into service/scriptTask and see how far I will get.
